### PR TITLE
[Doc] rewrite consensus section

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

I decided to remove most the previous consensus page, because they seem to be confusing.
Only a short overview section is added to replace the previous page. We can include more images later on. Not sure if we want very detailed docs here though, which can be hard to keep updated.

## Test plan 

n/a

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: docs/content/concepts/sui-architecture.mdx [hunk 1]
```diff
@@ -16,11 +16,12 @@ Life of a Transaction details the transitions that all transactions on Sui go th
 
 Go to [Life of a Transaction](./sui-architecture/transaction-lifecycle.mdx).
 
-## Mempool and Consensus Engines
+## Consensus
 
-Narwhal and Bullshark combine to form the basis of the Sui mempool and consensus engine. This topic offers a brief introduction to the technologies that keep the activities on Sui synchronized and secure. 
+Zooming in, every transaction is sequenced by consensus, where validators agree to the same order of execution of the transactions, even if a minority of them are down or want to cause disruptions!
```
Comment (akichidis) [inline docs/content/concepts/sui-architecture.mdx:21]: `or want to cause disruptions!` , maybe we can explicitly mention here the malicious actors, so something like

"or they are malicious and want to cause disruptions"

File: docs/content/concepts/sui-architecture/consensus.mdx [hunk 2]
```diff
@@ -1,109 +1,16 @@
...
+sidebar_label: Consensus 
+description: Overview of the Sui consensus.
 ---
 
-This is a brief introduction to [Narwhal](https://github.com/MystenLabs/sui/tree/main/narwhal), and [Bullshark](https://arxiv.org/abs/2209.05633), the high-throughput mempool and consensus engines offered by Mysten Labs. Sui uses Narwhal as the mempool and Bullshark as the consensus engine by default, to sequence transactions that require a total ordering, synchronize transactions between validators and periodically checkpoint the network's state.
+## Overview
 
-The names highlight that the components split the responsibilities of:
+The basic purpose of consensus in blockchains is to agree on the order and ensure the availability of transactions.
+Validator operators and users can think of Sui consensus as a black box: validators submit different user transactions to consensus concurrently, and the consensus outputs a consistent stream of transactions across all well-behaving validators.
```
Comment (akichidis) [inline docs/content/concepts/sui-architecture/consensus.mdx:10]: maybe `a consistent stream of deterministically ordered transactions...`

File: docs/content/concepts/sui-architecture/consensus.mdx [hunk 3]
```diff
@@ -1,109 +1,16 @@
...
 
-- ensuring the availability of data submitted to consensus = [Narwhal](https://arxiv.org/abs/2105.11827)
-- agreeing on a specific ordering of this data = [Bullshark](https://dl.acm.org/doi/abs/10.1145/3548606.3559361)
+Byzantine-fault tolerant (BFT) consensus protocols are a fascinating area of research. If you want a deeper understanding of the consensus black box, the next-generation consensus engine set to roll out in Sui is based on the **Mysticeti** protocol. It represents the cutting edge of research, achieving the optimal frontier of low latency and high throughput by:
+- Weaving blocks into directed acyclic graphs (DAGs), utilizing the full bandwidth of the network and providing censorship resistance.
+- Mysticeti DAGs in particular are uncertified, where it takes only three rounds of messages to commit blocks from the DAGs.
```
Comment (akichidis) [inline docs/content/concepts/sui-architecture/consensus.mdx:14]: I believe we can also mention pipelining & multi leader here as it's quite powerful and allow us to commit on every round. If we don't want to be too specific on pipelining maybe we can just refer to the universal commit rule - I believe we don't use that term any more but  something like:

`Applying a novel commit rule allowing multiple leaders to commit per round and on every round`

**Review by akichidis — APPROVED** [review]
Agreed on the more reduced consensus section. If we reach to a point where architecture and function wise consensus remains relatively unchanged then we could get back and add more context if desired.

**Review by ronny-mysten — CHANGES_REQUESTED** [review]
The consensus topic doesn't add any value in this form. "Click here to learn about consensus...tricked ya! it's a black box" :). What if we just point to the mysteceti research paper from sui-architecture.mdx for now? I can update this PR of what that might look like if y'all agree.

**Comment by mwtian** [comment]
> The consensus topic doesn't add any value in this form. "Click here to learn about consensus...tricked ya! it's a black box" :). What if we just point to the mysteceti research paper from sui-architecture.mdx for now? I can update this PR of what that might look like if y'all agree.

We can definitely just link to the research paper, but ideally there can be an accessible intro giving curious readers an intuition, similar to the Basecamp presentations. We can add more details here.

**Comment by ronny-mysten** [comment]
> ideally there can be an accessible intro giving curious readers an intuition.

Agreed. I was thinking that intro could be a part of the top page that links to that white paper. It would be better from a site structure point of view, though, to have the page remain. We need to make it more informative, though, and not use opinions or marketing. I'll suggest some changes and we can take it from there.

File: docs/content/concepts/sui-architecture/consensus.mdx (hunk 2 above)
Comment (mwtian) [reply docs/content/concepts/sui-architecture/consensus.mdx]: I think consistent means the stream needs to be deterministically ordered?

File: docs/content/concepts/sui-architecture/consensus.mdx (hunk 3 above)
Comment (mwtian) [reply docs/content/concepts/sui-architecture/consensus.mdx]: Added two points related to commit rule.

**Comment by mwtian** [comment]
> > ideally there can be an accessible intro giving curious readers an intuition.
> 
> Agreed. I was thinking that intro could be a part of the top page that links to that white paper. It would be better from a site structure point of view, though, to have the page remain. We need to make it more informative, though, and not use opinions or marketing. I'll suggest some changes and we can take it from there.

Most of the points are taken from the paper and I reworded them more. Feel free to give suggestions too.

**Review by ronny-mysten — APPROVED** [review]
I like this much better. Thanks!

**Comment by mwtian** [comment]
> I like this much better. Thanks!

Thanks for the help!

**Pull request merged**